browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -239,4 +239,39 @@ int OnBeforeURLRequest_AdBlockTPPreWork(const ResponseCallback& next_callback,
   return net::ERR_IO_PENDING;
 }
 
+int OnHeadersReceived_AdBlockCspWork(
+    const net::HttpResponseHeaders* response_headers,
+    scoped_refptr<net::HttpResponseHeaders>* override_response_headers,
+    GURL* allowed_unsafe_redirect_url,
+    const brave::ResponseCallback& next_callback,
+    std::shared_ptr<brave::BraveRequestInfo> ctx) {
+  std::string source_host = ctx->initiator_url.host();
+
+  if (ctx->resource_type == blink::mojom::ResourceType::kMainFrame ||
+      ctx->resource_type == blink::mojom::ResourceType::kSubFrame) {
+    std::string additional_directives;
+    const base::Optional<std::string> csp_directives =
+        g_brave_browser_process->ad_block_service()->GetCspDirectives(
```

Comment:
the rust core can be queried only on its own task runner, we can't just call it like this

Reply:
Fixed